### DISCUSSION TITLE
Media Manager Issues

### DISCUSSION THREAD

**Discussion by dgrammatiko** [joomla/joomla-cms#34927] [General]
This is not an exhaustive list, but these issues should be patched with a higher (or highest) priority:

- There needs to be a registry with the actual file content hash (updated per edit, etc), so there could be a proper cache invalidation for all the media files. The existing helper that removes the hash from the URL should also append a mediaVersion kinda thing as we do for the static assets as css,js.
- The folder tree view should have nice cascading (try creating multiple nested folders). Now it's broken
- The breadcrumbs should nicely hide parts for multiple nested folders. Now it's broken
- There should be a registry for the external adapters. Resolving each and every request.
- Have a way to create thumbs for the media manager views to reduce data consumption, increase performance, etc

**Comment by brianteeman** [comment]
Uploading a folder of images with drag and drop would be nice too

**Reply by dgrammatiko** [reply]
The platform has native support for this (well the existing drag and drop needs some changes): https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/webkitdirectory

**Reply by dgrammatiko** [reply]
Support is not great (all major desktop browser support it though) but progressive enhancement is always our friend: https://caniuse.com/input-file-directory

**Comment by bembelimen** [comment]
@dgrammatiko could you create an issue from your list, so it's not forgotten? Thanks

**Reply by dgrammatiko** [reply]
Well the list it's not exhaustive, someone needs to visit https://github.com/joomla-projects/media-manager-improvement/issues and figure out what was fixed and what not

**Comment by simbus82** [comment]
Hi! I have see that many improvements can be done here. I have opened this issue #35651 about the bad usability of the media manager.

1. Edits must be "recorded" when the user switch from one tab to another, and has to happen like an "apply" action (temporary), not like a "save" action.
2. A "Save as..." button is mandatory: 99% of users want to have the original image too, or save many versions of the original one.
3. If the user do a "save" action, he must be informed about that the original image will be overwritten, and the cms have to show a prompt with actual file name and the chance to edit the name of the file before saving.

It is essential to **fix points 2 and 3 as soon as possible** because they are creating damage to the assets of customers (users).

**Comment by brianteeman** [comment]
The media view should have pagination so that a user with a large number of images in a folder does not have to load them all every time they work on that image folder

**Reply by dgrammatiko** [reply]
I had some thoughts on this: https://github.com/joomla-projects/media-manager-improvement/issues/510#issue-295639911
